model/generate.go

```diff
@@ -582,10 +590,15 @@ func (g *GeneratedProject) findTasksAndVariantsWithSpecificActivations(requester
 		// Regardless of whether the build variant has batchtime, there may be tasks with different batchtime
 		batchTimeTasks := []string{}
 		for _, bvt := range bv.Tasks {
-			if isStepbackTask(g.Task, bv.Name, bvt.Name) {
-				stepbackInfo := specificStepbackInfo{task: bvt.Name}
-				res.stepbackTasks[bv.Name] = append(res.stepbackTasks[bv.Name], stepbackInfo)
-				continue // Don't consider batchtime/activation if we're stepping back this generated task
+			if g.Task.ActivatedBy == evergreen.StepbackTaskActivator {
```

Comment:
Can we add unit tests for this logic? It can be added to this [test](https://gist.github.com/hadjri/cbf32422c6b6c90045ba5924e8ea1648#file-generate_test-go-L1905) which calls `filterInactiveTasks` which propagates down to this function

Reply:
Nice catch! I added a unit test and I added some logic in the filter function to account for stepback tasks also being activated.